### DISCUSSION TITLE
How to add a dynamic set of keys to a new yaml file?

### DISCUSSION THREAD

**Discussion by gangofnuns** [mikefarah/yq#1253] [Q&A]
**Describe the bug**
I've been Googling for awhile, and can't find a way to do this.  
It seems like it should be simple, so I am missing something obvious. 

I want to create a yaml file with yaml keys defined dynamically at run time. 
Example: 
environment variables:  foo bar baz

Result in yaml:  
.this.foo: 
    -- stuff here
.this.bar: 
    -- stuff here
.this.baz:
    -- stuff here

The tool is very complicated, but I cannot seem to find an example of dynamic key creation
that would apply to this simple scenario.   Hoping there is an easy answer I've overlooked. 

**Input Yaml**
NONE

**Comment by gangofnuns** [comment]
reopening as v4 issue.

**Comment by gangofnuns** [comment]
Describe the bug
I've been Googling for awhile, and can't find a way to do this.
It seems like it should be simple, so I am missing something obvious.

I want to create a yaml file with yaml keys defined dynamically at run time.
Example:
environment variables: foo bar baz

Result in yaml:
.this.foo:
-- foo stuff here
.this.bar:
-- bar stuff here
.this.baz:
-- baz stuff here

The tool is very feature rich, but I cannot seem to find a simple example of dynamic key creation. 
Hoping there is an easy answer I've overlooked.

**Comment by gangofnuns** [comment]
- [v4]

**Comment by mikefarah** [comment]
I think you are looking for a variation of this https://mikefarah.gitbook.io/yq/operators/env-variable-operators#dynamic-key-lookup-with-environment-variable

You can create entries like:

```
foo=cat ./yq -n '.[strenv(foo)] = "meow"'
cat: meow
```

**Comment by mikefarah** [comment]
https://mikefarah.gitbook.io/yq/operators/env-variable-operators#dynamically-update-a-path-from-an-environment-variable

may also be of interest
